http-server/src/main/java/io/micronaut/http/server/binding/LocaleArgumentBinder.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.binding;
+
+import io.micronaut.core.convert.ArgumentConversionContext;
+import io.micronaut.core.type.Argument;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.bind.binders.TypedRequestArgumentBinder;
+import io.micronaut.http.server.util.HttpLocaleResolver;
+
+import javax.inject.Singleton;
+import java.util.Locale;
+import java.util.Optional;
+
+@Singleton
+public class LocaleArgumentBinder implements TypedRequestArgumentBinder<Locale> {
```

Comment:
Missing javadoc. Should this be public? Does it need to be an additional bean? Can it be registered programmatically in `DefaultRequestBinderRegistry`

Reply:
@jameskleeh what about this part of the comment: 

>  Should this be public? Does it need to be an additional bean? Can it be registered programmatically in DefaultRequestBinderRegistry

Reply:
It should be an additional bean so we don't need to tie the dependency on locale resolver to the binder registry